compiler/src/iree/compiler/Tools/iree_compile_lib.cc

```diff
@@ -161,6 +161,20 @@ int mlir::iree_compiler::runIreecMain(int argc, char **argv) {
       llvm::cl::desc("Split the input file into pieces and "
                      "process each chunk independently"),
       llvm::cl::init(false));
+  llvm::cl::opt<bool> listHalTargets(
+      "iree-hal-list-target-backends",
+      llvm::cl::desc(
+          "Lists all registered target backends for executable compilation."),
```

Comment:
I considered adding this flag to [`TargetRegistry.cpp`](https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Dialect/HAL/Target/TargetRegistry.cpp) or [`TargetBackend.cpp`](https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Dialect/HAL/Target/TargetBackend.cpp), but it didn't quite fit in, without a bit of refactoring. Is this location (`iree_compile_lib.cc`) okay?